data-processor/src/test/groovy/io/micronaut/data/processor/sql/BuildQuerySpec.groovy

```diff
@@ -150,6 +151,47 @@ interface MealRepository extends CrudRepository<Meal, Long> {
 
     }
 
+    void "test join query on collection via association entity"() {
+        given:
+        def repository = buildRepository('test.BookRepository', """
+import io.micronaut.context.annotation.Executable;
+import io.micronaut.data.annotation.Join;
+import io.micronaut.data.jdbc.annotation.JdbcRepository;
+import io.micronaut.data.model.query.builder.sql.Dialect;
+import io.micronaut.data.repository.GenericRepository;
+import io.micronaut.data.tck.entities.Author;
+import io.micronaut.data.tck.entities.Book;
+
+import java.util.Optional;
+
+@JdbcRepository(dialect = Dialect.H2)
+interface BookRepository extends GenericRepository<Book, Long> {
+
+    @Join(value = "author.books", type = Join.Type.LEFT_FETCH)
+    public abstract Author findAuthorById(@Id Long id);
+
+    public abstract List<Book> findAuthorBooksById(@Id Long id);
```

Comment:
I would expect this to fail because "author.books" is not joined

Reply:
Explained above, it gets added via projection (author) and its joins